### PR TITLE
modify the poster picking so we don't run on the same poster twice du…

### PR DESCRIPTION
**Pull request by Valentin-Morette**
…ring the game

### CODE REVIEW

File: frontend/src/context/ApiCallsContext.jsx [hunk 1]
```diff
@@ -39,6 +37,7 @@ export function ApiCallsContextProvider({ children }) {
           .sort(() => Math.random() - 0.5),
       });
     });
+    movieIdArray.push(movieIdArray.splice(0, 1));
```
Comment (nibor51) [inline frontend/src/context/ApiCallsContext.jsx:40]: With this we lose the notion of randomness during the game. You have to move four cards and not one! Try with a state and by making a shift of 4 "notches" in your table for each question.

**Review by nibor51 — CHANGES_REQUESTED** [review]

**Review by nibor51 — CHANGES_REQUESTED** [review]
I don't see the state, nor the shift

File: frontend/src/pages/Play.jsx [hunk 2]
```diff
@@ -46,15 +46,16 @@ export default function Play() {
   };
 
   function nextLevel() {
-    pickMovie(movieIdArray);
+    pickMovie(movieIdArray, (count + 1) * 4);
```
Comment (nibor51) [inline frontend/src/pages/Play.jsx:49]: move the multiplication into the function

File: frontend/src/pages/Play.jsx [hunk 3]
```diff
@@ -46,15 +46,16 @@ export default function Play() {
...
     setCount(count + 1);
     reset();
     start();
     setIsAnswerActive(false);
   }
 
   useEffect(() => {
-    pickMovie(movieIdArray);
+    movieIdArray.sort(() => Math.random() - 0.5);
+    pickMovie(movieIdArray, count * 4);
```
Comment (nibor51) [inline frontend/src/pages/Play.jsx:58]: same one

**Review by nibor51 — CHANGES_REQUESTED** [review]
currently in your function you never use the first 4 films.
To fix this, set count to zero and don't multiply it.

Hint : In the JSX, there you can add one to count to see it

File: frontend/src/pages/Play.jsx [hunk 4]
```diff
@@ -46,15 +46,16 @@ export default function Play() {
   };
 
   function nextLevel() {
-    pickMovie(movieIdArray);
+    pickMovie(movieIdArray, count + 1);
```
Comment (rocambille) [inline frontend/src/pages/Play.jsx:49]: count is starting at 1 and you're adding one here: you're droping the eight first elements

**Review by rocambille — CHANGES_REQUESTED** [review]

**Pull request merged**